src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp

```diff
@@ -28,18 +34,45 @@ TEST_F(KickoffFriendlyPlayTest, test_kickoff_friendly_play)
     setRefereeCommand(RefereeCommand::NORMAL_START, RefereeCommand::PREPARE_KICKOFF_US);
 
     std::vector<ValidationFunction> terminating_validation_functions = {
-        // This will keep the test running for 9.5 seconds to give everything enough
-        // time to settle into position and be observed with the FullSystemGUI
-        // TODO: Implement proper validation
-        // https://github.com/UBC-Thunderbots/Software/issues/1396
         [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
-            while (world_ptr->getMostRecentTimestamp() < Timestamp::fromSeconds(9.5))
-            {
-                yield();
-            }
+            // Two friendly robots near the half line setting up for offense
+            Rectangle robotsOffensiveRect(Point(-1.75, 2.5), Point(-1.5, -2.5));
+            robotInPolygon(1, robotsOffensiveRect, world_ptr, yield);
+            robotInPolygon(5, robotsOffensiveRect, world_ptr, yield);
+
+
+            // Two Friendly robots defending the exterior of defense box and one goalie
+            Rectangle robotsDefensiveRect(Point(-3.2, 1.1), Point(-3.5, -1.1));
+            robotInPolygon(0, robotsDefensiveRect, world_ptr, yield);
+            robotInPolygon(2, robotsDefensiveRect, world_ptr, yield);
+            robotInPolygon(3, robotsDefensiveRect, world_ptr, yield);
+
+            // Robot 4 is the only robot allowed to be in the center circle and start
+            // the kickoff
+            robotInCenterCircle(4, world_ptr, yield);
+            robotReceivedBall(4, world_ptr, yield);
+            robotKickedBall(4, Angle::zero(), world_ptr, yield);
         }};
 
-    std::vector<ValidationFunction> non_terminating_validation_functions = {};
+    std::vector<ValidationFunction> non_terminating_validation_functions = {
+        [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
+            Robot robot0 = world_ptr->friendlyTeam().getRobotById(0).value();
+            Robot robot1 = world_ptr->friendlyTeam().getRobotById(1).value();
+            Robot robot2 = world_ptr->friendlyTeam().getRobotById(2).value();
+            Robot robot3 = world_ptr->friendlyTeam().getRobotById(3).value();
+            Robot robot5 = world_ptr->friendlyTeam().getRobotById(5).value();
+
+            robotInFriendlyHalf(robot0, world_ptr, yield);
+            robotInFriendlyHalf(robot1, world_ptr, yield);
+            robotInFriendlyHalf(robot2, world_ptr, yield);
+            robotInFriendlyHalf(robot3, world_ptr, yield);
+            robotInFriendlyHalf(robot5, world_ptr, yield);
+            robotNotInCenterCircle(robot0, world_ptr, yield);
+            robotNotInCenterCircle(robot1, world_ptr, yield);
+            robotNotInCenterCircle(robot2, world_ptr, yield);
+            robotNotInCenterCircle(robot3, world_ptr, yield);
+            robotNotInCenterCircle(robot5, world_ptr, yield);
```

Comment:
can we use some sort of for loop/ pass a vector of robot ids in?

Reply:
Fixed